overlord/snapstate/backend/link.go

```diff
@@ -184,6 +185,25 @@ func (b Backend) LinkSnap(info *snap.Info, dev snap.Device, linkCtx LinkContext,
 	return rebootInfo, nil
 }
 
+func componentLinkPath(mountDir string) (string, error) {
+	// The symlink to current is the mount with the stripped revision
+	revStart := strings.LastIndex(mountDir, "_")
+	if revStart == -1 {
+		return "", fmt.Errorf("internal error: no revision in %q", mountDir)
+	}
+	return mountDir[0:revStart], nil
+}
+
+func (b Backend) LinkComponent(cpi snap.ContainerPlaceInfo) error {
+	mountDir := cpi.MountDir()
+	linkPath, err := componentLinkPath(mountDir)
```

Comment:
hmm, what if we extend snap.ContainerPlaceInfo to add eg. `ActiveSymlinkName()` which would return `/snap/<name>/current` for snaps and `/snap/<name>/components/<comp>` for components?

Reply:
I've been thinking about this, but I'm not fully convinced as strictly speaking `ContainerPlaceInfo` does not need to know what is an active snap/component or how those reference to the containers.